servers/movie_writer/movie_writer_ogv.h

```diff
@@ -0,0 +1,144 @@
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include "servers/audio_server.h"
+#include "servers/movie_writer/movie_writer.h"
+
+#include <theora/theoraenc.h>
+#include <vorbis/codec.h>
+#include <vorbis/vorbisenc.h>
+
+class MovieWriterOGV : public MovieWriter {
+	GDCLASS(MovieWriterOGV, MovieWriter)
+
+	uint32_t mix_rate = 48000;
+	AudioServer::SpeakerMode speaker_mode = AudioServer::SPEAKER_MODE_STEREO;
+	String base_path;
+	uint32_t frame_count = 0;
+	uint32_t fps = 0;
+	uint32_t audio_ch = 0;
+	uint32_t audio_frames = 0;
+
+	Ref<FileAccess> f;
+
+	// Bitrate target for Vorbis audio.
+	int audio_bitrate = 0;
+
+	// Vorbis quality -0.1 to 1 (-0.1 yields smallest files but lowest fidelity; 1 yields highest fidelity but large files. '0.2' is a reasonable default).
+	float audio_quality = 0.2;
+
+	// VP3 strict compatibility.
+	int vp3_compatible = 0;
+
+	// Bitrate target for Theora video.
+	int video_bitrate = 0;
+
+	// Theora quality selector from 0 to 1.0 (0 yields smallest files but lowest video quality. 1.0 yields highest fidelity but large files).
+	float video_quality = 0.75;
+
+	// Video stream keyframe frequency (one every N frames).
+	ogg_uint32_t keyframe_frequency = 64;
```

Comment:
This is hardcoded to `64` currently. Should we use a different value (to improve the quality/size ratio) or add a project setting to configure it?

cc @DeeJayLSP

Reply:
I think that's a good default, although a lower value might improve the quality in some cases, and the granularity on seeks. It could be a good candidate for a setting.